modules/stochastic_tools/src/state_sim/common/StateSimBitset.C

```diff
@@ -0,0 +1,156 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "StateSimBitset.h"
+#include "MooseApp.h"
+#include <vector>
+
+const int NUMBITS = 32;
+const int NUMBITS_MIN1 = 31;
+
+StateSimBitset::StateSimBitset(int length)
+  : _m_array(),
+    _m_length(length)
+{
+  mooseAssert((length > 0), "StateSimBitset - length must be >= 0");
+
+  unsigned long slots = (unsigned long)((length + NUMBITS_MIN1) / NUMBITS);
+  while (_m_array.size() < slots)
+  {
+    _m_array.push_back(0);
```

Comment:
Use `resize()` or `reserve` here. Pushing back repeatedly may cause several reallocation and unecessary copies. There's an `::assign` method you should look at too.